java/java.mx.project/src/org/netbeans/modules/java/mx/project/CoreSuite.java

```diff
@@ -0,0 +1,732 @@
+                return distributions;
+            }
+        };
+    }
+
+    private static MxDistribution createMxDistribution(
+        List<String> dependencies,
+        List<String> distDependencies,
+        List<String> exclude,
+        List<String> strip
+    ) {
+        return new MxDistribution() {
+            @Override
+            public List<String> dependencies() {
+                return dependencies;
+            }
+
+            @Override
+            public List<String> distDependencies() {
+                return distDependencies;
+            }
+
+            @Override
+            public List<String> exclude() {
+                return exclude;
+            }
+
+            @Override
+            public List<String> strip() {
+                return strip;
+            }
+        };
+    }
+
+    private static MxProject createMxProject(
+        List<String> annotationProcessors,
+        List<String> dependencies,
+        String dir,
+        String javaCompliance,
+        List<String> sourceDirs,
+        String subDir
+    ) {
+        return new MxProject() {
+            @Override
+            public String dir() {
+                return dir;
+            }
+
+            @Override
+            public String subDir() {
+                return subDir;
+            }
+
+            @Override
+            public List<String> sourceDirs() {
+                return sourceDirs;
+            }
+
+            @Override
+            public List<String> dependencies() {
+                return dependencies;
+            }
+
+            @Override
+            public List<String> annotationProcessors() {
+                return annotationProcessors;
+            }
+
+            @Override
+            public String javaCompliance() {
+                return javaCompliance;
+            }
+        };
+    }
+
+    private static MxLibrary createMxLibrary(
+        List<String> dependencies,
+        Map<String, MxLibrary.Arch> osArch,
+        String path,
+        String sha1,
+        List<String> urls
+    ) {
+        return new MxLibrary() {
+            @Override
+            public String sha1() {
+                return sha1;
+            }
+
+            @Override
+            public List<String> urls() {
+                return urls;
+            }
+
+            @Override
+            public Map<String, MxLibrary.Arch> os_arch() {
+                return osArch;
+            }
+
+            @Override
+            public List<String> dependencies() {
+                return dependencies;
+            }
+
+            @Override
+            public String path() {
+                return path;
+            }
+
+        };
+    }
+
+    private static MxLibrary.Arch createArch(MxLibrary amd64) {
+        return new MxLibrary.Arch() {
+            @Override
+            public MxLibrary amd64() {
+                return amd64;
+            }
+        };
+    }
+
+    static final MxSuite CORE_5_279_0;
+    static {
```

Comment:
I wonder - can we parse this at runtime from something? Like, maybe, mx/suite.py? It feels weird to have this information hardcoded.

Reply:
That would require `mx.mx/suite.py` being around. While `mx` is needed for execution/debugging of tests, it is not needed for opening the projects and navigating the sources right now. However, to properly resolve "standard mx libraries" we need to have information about them. That is what this file provides.

Having the information here improves out of box experience. One can `git clone https://github.com/oracle/graal` and immediately open and inspect the sources.

Of course, when new version of `mx` introduces new "standard libraries", we might need to release an update to this `CoreSuite` data. Possibly we could use this as default/fallback. Once `mx` is found/specified we could switch to its `mx.mx/suite.py` definitions.